app/travis_scripts/build-docker-env.sh

```diff
@@ -12,11 +14,32 @@ sudo ifconfig docker0 10.200.10.1/24
 git clone https://github.com/streamr-dev/streamr-docker-dev.git
 # start everything except eth watcher
 streamr-docker-dev/streamr-docker-dev/bin.sh start 5
-# wait for e&e to be up
-while true; do http_code=$(curl -s -o /dev/null -I -w "%{http_code}" http://localhost:8081/streamr-core/login/auth); if [ $http_code = 200 ]; then echo "EE up and running"; break; else echo "EE not receiving connections"; sleep 5s; fi; done
-# data-api sometimes fails to boot properly (??!) if services start in wrong order
+
+RETRIES=30;
+RETRY_DELAY=5s;
+
+# wait for E&E to come up
+waitFor $RETRIES $RETRY_DELAY checkHTTP "engine-and-editor" 200 http://localhost:8081/streamr-core/login/auth
+
+# exit if E&E never comes up
+if [ $? -eq 1 ] ; then
+    echo "engine-and-editor never up";
+    exit 1;
+fi
+
 streamr-docker-dev/streamr-docker-dev/bin.sh restart data-api # let's restart it for good measure (?!)
-# wait for data-api to be up
-while true; do http_code=$(curl -s -o /dev/null -I -w "%{http_code}" http://localhost:8890/); if [ $http_code = 404 ]; then echo "Data-api up and running"; break; else echo "Data API not receiving connections"; sleep 5s; fi; done
-# add a few more waits
-sleep 10
+
+# wait briefly for data-api to come up. it probably needs restarting again.
+waitFor 15 3s checkHTTP "data-api" 401 http://localhost:8890/
+
+# try restarting data-api again if still not up
+if [ $? -eq 1 ] ; then
+    streamr-docker-dev/streamr-docker-dev/bin.sh restart data-api
```

Comment:
Would it make sense to actually spit out the contents of Data API docker logs at this point (before restarting)? It could be useful for diagnosing the underlaying issue. You can do it with `docker-compose logs --tail=200 data-api`.